### ISSUE TITLE
amp-lightbox-gallery: Non-text content has insufficient color contrast

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31158]
## WCAG Level
Level AA

## Priority
High

## Type
Color

## Pages/screens/components affected
* [amp-lightbox-gallery](https://amp.dev/documentation/components/amp-lightbox-gallery/)

## Description
An important design consideration is to make sure there is enough contrast between the color of meaningful non-text elements (such as controls denoted purely through graphical elements, icons used as the only visible indicator for errors, charts/graphs, meaningful borders/outlines used to denote a selected item) and their background. Good color contrast ensures that all sighted users, including those who have a condition that affects their ability to perceive color, are able to see/understand meaningful non-text content.

Depending on the browser window dimensions, and the dimensions of the images/slides, the "Gallery" and "Close" controls of `amp-lightbox-gallery` can appear overlaid on the slide content itself. Depending on the slide content, this can result in the controls having low contrast (or, if the image is pure white, no contrast at all).

![White lightbox controls shown over a light yellow slide, with very low contrast](https://user-images.githubusercontent.com/54933393/99181210-ef2de380-2724-11eb-8afc-42497c195d0d.png)

## User impact
Without sufficient contrast, content can be difficult to perceive and therefore difficult to identify. In extreme cases, content with low contrast may not be visible at all to people with color perception impairments.

In this case, users with impaired color perception may be unable to clearly perceive the lightbox controls.

## Required solution
Make sure that non-text content has a color contrast ratio of at least 3:1 against its background and the content that is adjacent to it.

Elements that are disabled or inactive, and elements that use the browser's default styles, are exempt.

This solution must be applied to all instances of the issue identified within the test sample, then applied to all other instances of the same issue identified throughout the rest of the website.

## Implementation guidance
Currently, the lightbox controls are simply white SVG background images, with no further styling provided.

```
<div role="button" class="i-amphtml-lbg-button" data-action="close" aria-label="Close"></div>
...
<div role="button" class="i-amphtml-lbg-button" data-action="gallery" aria-label="Gallery"></div>
```

```
.i-amphtml-lbg-button[data-action=close]:after {
  background-image: url(data:image/svg+xml;charset=utf-8,...);
}

...

.i-amphtml-lbg-button[data-action=gallery]:after {
  background-image: url(data:image/svg+xml;charset=utf-8,...);
}
```

We recommend adding additional styling to either the button itself, or the SVG, to provide sufficient contrast against bright/white backgrounds. For instance, the SVG elements could be modified to include a subtle dark/black outline, or the button could have a semi-transparent (but still sufficiently dark, against a white background) black/gray background color.

![Example of a lightbox button with a semi-transparent background](https://user-images.githubusercontent.com/54933393/99181222-f94fe200-2724-11eb-96d5-76a6a7f2642c.png)

```
.i-amphtml-lbg-button {
  ...
  background: hsla(0,0%,0%,0.42);
  ...
}
```

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Check the contrast of all non-text content using a tool such as the [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/).
2. Check that the contrast level between the element and its adjacent content is at least 3:1.

### Definition of done
Complete all of these tasks before closing this issue or indicating it is ready for retest:
* All issues identified within the test sample have been resolved.
* The rest of the website has been tested for the same issue.
* All issues identified throughout the rest of the website  have been resolved or filed as new issues.

## Related standards
* [WCAG 2.1 Success Criterion 1.4.11 Non-text Contrast (Level AA)](https://www.w3.org/TR/WCAG21/#non-text-contrast)

## More information
* [TPG: Colour Contrast Analyser](https://developer.paciellogroup.com/resources/contrastanalyser/)
* [WebAIM: Contrast and Color Accessibility - 1.4.11 Non-text Contrast](https://webaim.org/articles/contrast/#sc1411)
* [WebAIM: Color blindness](https://webaim.org/articles/visual/colorblind)

## Test data
Test date: November 2020
